lib/pkg.ml

```diff
@@ -246,21 +225,24 @@ let distrib_uri_of_dev_repo p =
   opam_field_hd p "dev-repo" >>= function
   | None -> Ok None
   | Some dev_repo ->
-      let dev_repo =
-        match String.cut ~sep:"git@github.com:" dev_repo with
-        | Some ("", path) -> "https://github.com/" ^ chop_ext path
-        | _ -> (
-            match String.cut ~sep:"git+ssh://git@github.com/" dev_repo with
-            | Some ("", path) -> "https://github.com/" ^ chop_ext path
-            | _ -> chop_git_prefix (chop_ext dev_repo) )
-      in
-      path_of_distrib p >>| fun path -> Some (uri_append dev_repo path)
+      path_of_distrib p >>= fun path ->
+      Fpath.of_string path >>= fun path ->
+      Github_uri.Repo.of_string dev_repo >>| fun { user; repo; _ } ->
+      let scheme = Github_uri.WWW_scheme.HTTPS in
+      Some (Github_uri.Distrib.make ~user ~repo ~scheme ~path)
 
 let distrib_uri_of_homepage p =
-  opam_homepage_sld p >>= function
+  opam_homepage p >>= function
   | None -> Ok None
-  | Some (uri, _) ->
-      path_of_distrib p >>| fun path -> Some (uri_append uri path)
+  | Some s -> (
+      path_of_distrib p >>= fun path' ->
+      Fpath.of_string path' >>= fun path' ->
+      Github_uri.Homepage.of_string s >>| fun { user; repo; scheme; path } ->
+      match path with
+      | Some path ->
+          let path = Fpath.append path path' in
+          Some (Github_uri.Distrib.make ~user ~repo ~scheme ~path)
+      | None -> Some (Github_uri.Distrib.make ~user ~repo ~scheme ~path:path') )
 
 let distrib_uri ?(raw = false) p =
```

Comment:
Is there any reason why this is still exposed and used elswhere in the code considering there has to be a github uri for this function to return `Ok _` ?